test/gtest/tensor_layout_util.hpp

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef TENSOR_LAYOUT_UTIL_HPP
+#define TENSOR_LAYOUT_UTIL_HPP
+
+#include <miopen/miopen.h>
+
+#include <string>
+
+inline std::string tensor_layout_to_string(miopenTensorLayout_t layout)
+{
+    switch(layout)
+    {
+    case miopenTensorNCHW: return "NCHW";
+    case miopenTensorNCDHW: return "NCDHW";
+    case miopenTensorNHWC: return "NHWC";
+    case miopenTensorNDHWC: return "NDHWC";
+    default: MIOPEN_THROW("Unsupported tensor layout");
+    }
+}
```

Comment:
[Note] We already have `TensorDescriptor::GetLayout_str()`. There is no static method there, it might be worth adding.